Sources/PostgresNIO/PostgresRequest.swift

```diff
@@ -1,5 +1,8 @@
 import Logging
 
+/// Protocol to encapsulate a function call on the Postgres server
+///
+/// This protocol is deprecated going forward. 
```

Comment:
Should we mark it with `@deprecated`?

Reply:
We don't want to mark anything `deprecated` in this pr to change as little as possible in the public API surface. However we should definitely change this with subsequent prs.